lib/tools/adb-commands.js

```diff
@@ -78,28 +78,26 @@ methods.initBundletool = async function initBundletool () {
  *                  Android Lollipop. The result of this method is cached, so all the further
  * calls return the same value as the first one.
  */
-methods.getApiLevel = async function getApiLevel () {
-  if (!_.isInteger(this._apiLevel)) {
-    try {
-      const strOutput = await this.getDeviceProperty('ro.build.version.sdk');
-      let apiLevel = parseInt(strOutput.trim(), 10);
-
-      // Temp workaround. Android Q beta emulators report SDK 28 when they should be 29
-      if (apiLevel === 28 && (await this.getDeviceProperty('ro.build.version.release')).toLowerCase() === 'q') {
-        log.debug('Release version is Q but found API Level 28. Setting API Level to 29');
-        apiLevel = 29;
-      }
-      this._apiLevel = apiLevel;
-      log.debug(`Device API level: ${this._apiLevel}`);
-      if (isNaN(this._apiLevel)) {
-        throw new Error(`The actual output '${strOutput}' cannot be converted to an integer`);
-      }
-    } catch (e) {
-      throw new Error(`Error getting device API level. Original error: ${e.message}`);
+methods.getApiLevel = _.memoize(async function getApiLevel () {
```

Comment:
I don't think this is a good idea. We must do memoization based on ADB instance rather than statically based. Why you don't like the previous approach

Reply:
The previous approach was fine I can change it back. I just liked using memoize because it better expresses what the code is doing.